src/il/tos.nim

```diff
@@ -178,35 +181,37 @@ proc `$`*(self: Pattern): string =
     of PatternKind.UnderScore:
         "_"
 proc `$`*(self: IdentDef): string =
-    case self.kind
-    of DefKind.Def:
-        let
-            pat = $self.pat
-            typ = if self.typ.isNone: "" else: fmt": {self.typ.get}"
-            default = if self.default.isNone: "" else: fmt" = {self.default.get}"
-            docStr = if self.docStr.isNone: "" else: fmt" #{self.docStr.get}"
-        pat & typ & default & docStr
-    of DefKind.Comment:
-        "#" & self.comment
+    let
+        pat = $self.pat
+        typ = if self.typ.isNone: "" else: fmt": {self.typ.get}"
+        default = if self.default.isNone: "" else: fmt" = {self.default.get}"
+        docStrs = self.comments.filterIt(it.isDoc)
+        comments = self.comments.filterIt(not it.isDoc)
+    assert docStrs.len <= 1
+    let
+        docStr = if docStrs.len != 0: fmt" {docStrs[0]}" else: ""
+        comment = if comments.len != 0: "\n" & fmt"{comments}" else: ""
+    pat & typ & default & docStr & comment
```

Comment:
use fmt function, like fmt"{pat}{typ}{docStr}{comment}"

Reply:
It is more intuitive to use the `fmt` function

Reply:
i see